tests/unit/s2n_connection_serialize_test.c

```diff
@@ -473,6 +490,271 @@ int main(int argc, char **argv)
+            EXPECT_OK(s2n_send_and_recv_test(server_conn, client_conn));
+            EXPECT_OK(s2n_send_and_recv_test(client_conn, server_conn));
+
+            uint8_t buffer[S2N_SERIALIZED_CONN_TLS12_SIZE] = { 0 };
+            if (serialized_mode[i] == S2N_CLIENT) {
+                EXPECT_SUCCESS(s2n_connection_serialize(client_conn, buffer, sizeof(buffer)));
+                /* Free old client */
+                EXPECT_OK(s2n_connection_ptr_free(&client_conn));
+
+                client_conn = s2n_connection_new(S2N_CLIENT);
+                EXPECT_NOT_NULL(client_conn);
+                EXPECT_SUCCESS(s2n_connection_deserialize(client_conn, buffer, sizeof(buffer)));
+            } else {
+                EXPECT_SUCCESS(s2n_connection_serialize(server_conn, buffer, sizeof(buffer)));
+                /* Free old server */
+                EXPECT_OK(s2n_connection_ptr_free(&server_conn));
+
+                server_conn = s2n_connection_new(S2N_SERVER);
+                EXPECT_NOT_NULL(server_conn);
+                EXPECT_SUCCESS(s2n_connection_deserialize(server_conn, buffer, sizeof(buffer)));
+            }
+
+            /* Wipe and re-initialize IO pipes */
+            EXPECT_SUCCESS(s2n_stuffer_wipe(&io_pair.client_in));
+            EXPECT_SUCCESS(s2n_stuffer_wipe(&io_pair.server_in));
+            EXPECT_OK(s2n_connections_set_io_stuffer_pair(client_conn, server_conn, &io_pair));
+
+            /* Client initiates key update */
+            EXPECT_SUCCESS(s2n_connection_request_key_update(client_conn, S2N_KEY_UPDATE_NOT_REQUESTED));
+
+            /* Sending and receiving is successful after key update */
+            EXPECT_OK(s2n_send_and_recv_test(client_conn, server_conn));
+            EXPECT_EQUAL(client_conn->send_key_updated, 1);
+            EXPECT_EQUAL(server_conn->recv_key_updated, 1);
+
+            /* Server initiates key update */
+            EXPECT_SUCCESS(s2n_connection_request_key_update(server_conn, S2N_KEY_UPDATE_NOT_REQUESTED));
+
+            /* Sending and receiving is successful after key update */
+            EXPECT_OK(s2n_send_and_recv_test(server_conn, client_conn));
+            EXPECT_EQUAL(client_conn->recv_key_updated, 1);
+            EXPECT_EQUAL(server_conn->send_key_updated, 1);
+
+            EXPECT_SUCCESS(s2n_connection_free(client_conn));
+            EXPECT_SUCCESS(s2n_connection_free(server_conn));
+        };
+    };
+
+    /* Self talk: Test interaction between renegotiation and serialization. Renegotiation is not available
+     * after serialization/deserialization. The information needed to perform renegotiation (i.e.
+     * client/server finished verify data and secure renegotiation flag) isn't stored during the
+     * serialization process and therefore isn't available post-deserialization.
+     * We could add that data to the serialized struct in the future, but for now, the user will get
+     * an error if they attempt to perform renegotiation after serialization. */
```

Comment:
Whitespace is your friend and helps readability.
```suggestion
    /* Self talk: Test interaction between renegotiation and serialization.
     *
     * Renegotiation is not available after serialization/deserialization.
     * The information needed to perform renegotiation (i.e. client/server finished verify data
     * and secure renegotiation flag) isn't stored during the serialization process and therefore
     * isn't available post-deserialization.
     * 
     * We could add that data to the serialized struct in the future, but for now, the user will get
     * an error if they attempt to perform renegotiation after serialization.
     */
```

Reply:
I really need to work on my whitespace game. It looks so much better afterwards 😭.